### PR TITLE
fix(react-native): add View as a container for react-native

### PR DESCRIPTION
**Pull request by bobylito**
The container added in a previous patch in the <InstantSearch /> was not working because strings can't be used in React-Native for createElement. It introduced a bug that [some users found ](https://stackoverflow.com/questions/41064940/using-algolia-react-instantsearch-with-react-native)

In order to fix it and not add a dependency to RN, it fallbacks to using the previous behaviour when not in DOM.

### CODE REVIEW

**Comment by vvo** [comment]
> In order to fix it and not add a dependency to RN, it fallbacks to using the previous behaviour when not in DOM.

What about we do not add any dependency and assume people willing to build instantsearch with react native do have react-native installed? This is what we are doing for react-dom, nowhere we have "react-dom" as a dep (not in peer, not in deps).

So that we could keep the same behavior. Would that work?

Ultimately we may have three packages:

react-intantsearch-dom
react-instantsearch-native
react-instantsearch

Just like what facebook does. So we can add any peer/dep we want that makes sense given the use case.

**Comment by vvo** [comment]
Cannot even right a proper test for react-native as this:

```js
/* eslint-env jest */

import 'react-native';
import React from 'react';
// import {InstantSearch} from './native.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
const algoliaClient = {search() { console.log('test'); }};

it('renders correctly', () => {
  const Root = 'View';
  const App = () => <Root/>;

  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
```

Just works, might be Jest being too optimistic

**Comment by bobylito** [comment]
I didn't know we didn't have a hard dependency on react-dom 🤔 since it works, like you said, then it's nice because it will be consistent across all platforms👌 making the changes tomorrow 🤓

**Comment by mthuret** [comment]
The change are:

- we passe directly View as a root for the react-native InstantSearch 
- add react-native example folder + simplification of our infinitsearch example

File: packages/react-instantsearch/examples/reactnative/App.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    flexDirection: 'row',
+    alignItems: 'center',
+    backgroundColor: 'white',
+  },
+});
+
+export default React.createClass({
+  render() {
+    return (
+      <View style={styles.maincontainer}>
```
Comment (bobylito) [inline packages/react-instantsearch/examples/reactnative/App.js]: So the View in a View works?

File: packages/react-instantsearch/examples/reactnative/App.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        >
+          <SearchBox/>
+          <Hits/>
+        </InstantSearch>
+      </View>
+    );
+  },
+});
+
+const SearchBox = connectSearchBox(React.createClass({
```
Comment (bobylito) [inline packages/react-instantsearch/examples/reactnative/App.js]: React.createClass is kinda old school. People tend to favor ES6 class or functions #notBlocking :)

File: packages/react-instantsearch/examples/reactnative/App.js (hunk 1 above)
Comment (mthuret) [inline packages/react-instantsearch/examples/reactnative/App.js]: Yes after cleaning up a lot of Views of the Example, everything went fine 👍

File: packages/react-instantsearch/examples/reactnative/App.js (hunk 2 above)
Comment (mthuret) [inline packages/react-instantsearch/examples/reactnative/App.js]: Doing it right now :)

File: packages/react-instantsearch/examples/reactnative/.gitignore [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# OSX
```
Comment (vvo) [inline packages/react-instantsearch/examples/reactnative/.gitignore]: Is this the default gitignore generated by react-native? There's a lot in it.

File: packages/react-instantsearch/examples/reactnative/App.js (hunk 1 above)
Comment (vvo) [inline packages/react-instantsearch/examples/reactnative/App.js]: NICE

**Pull request merged**